### ISSUE TITLE
gnutls version check is incorrect

### ISSUE THREAD

**Issue by davide125** [systemd/systemd#9403]
**systemd version the issue has been seen with**
> 239

**Used distribution**
 > CentOS 7

**Expected behaviour you didn't see**
 The build should either fail and clarify the missing dependency, or disable the DNS-over-TLS feature 

**Unexpected behaviour you saw**
 ```
../src/resolve/resolved-dns-transaction.c:17:27: fatal error: gnutls/socket.h: No such file or directory
 #include <gnutls/socket.h>
                           ^
compilation terminated.
```

**Steps to reproduce the problem**
When DNS-over-TLS is enabled, `src/resolve/resolved-dns-transaction.c` includes `gnutls/socket.h`, which was added in https://github.com/gnutls/gnutls/commit/0f67ccecddbe6533b51e74c955a4629654e79ae3#diff-d468b0837d8f0c1f2c70b0252c105a6a and is not available in gnutls 3.1.4 (which is the one we test for in meson). I noticed this because CentOS 7 ships with 3.3.26, which is also missing the include.

**Comment by davide125** [comment]
To clarify, the build works fine with `dns-over-tls=false` in this case, it only breaks when it's set to auto or true.

**Comment by yuwata** [comment]
Fix is waiting in #9410. But I have not checked the PR certainly works... Could you check the PR?
